### PR TITLE
For #282: Change main deck on hero change

### LINKED ISSUE #282: DeckBuilder: On Main Card pane fill paged with all Matching Class Cards
Main pane should display all cards in the Library that match the class of the Current Character.

Next/Previous buttons should be hidden if there is no Next/Previous page. 

Create a bunch of test cards for at least one class for visualizing this feature.

Estimate: 2/units

### PR DESCRIPTION
**Pull request by paulodamaso**
Closes #282: Created code which changes deck on hero change

This is an initial implementation. We have some design problems which impedes the hero selection when clicking on each portrait (I've left a puzzle to correct it).

Also, in my opinion this issue is too complex to be finished in just two time units, so I've just made the card filtering part by hero

### CODE REVIEW

File: src/ProjectNeon/Assets/State/Templates/DeckBuilderState.cs [hunk 1]
```diff
@@ -8,15 +9,33 @@ public class DeckBuilderState : ScriptableObject
...
+        get => currentHero;
+        set => currentHero = value;
+    }
+
     private Deck current;
     public Deck Current()
     {
-        return this.decks.Decks[0];
+        Deck deck = new Deck();
+        _library.UnlockedCards.ForEach(
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/State/Templates/DeckBuilderState.cs]: Should never auto-build a deck for the player.

File: src/ProjectNeon/Assets/State/Templates/DeckBuilderState.cs [hunk 2]
```diff
@@ -8,15 +9,33 @@ public class DeckBuilderState : ScriptableObject
     [SerializeField] private PartyDecks decks;
     [SerializeField] private Hero currentHero;
     [SerializeField] private GameEvent onCurrentDeckChanged;
-    
+    [SerializeField] private Library _library;
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/State/Templates/DeckBuilderState.cs]: No underscope here as per PR Policy.

**Comment by paulodamaso** [comment]
@SilasReinagel Corrections made, please take a look

File: src/ProjectNeon/Assets/State/Templates/DeckBuilderState.cs [hunk 3]
```diff
@@ -8,19 +9,43 @@ public class DeckBuilderState : ScriptableObject
...
+        set => currentHero = value;
+    }
+
     private Deck current;
     public Deck Current()
     {
         return this.decks.Decks[0];
     }
 
+    public Deck currentHeroDeck()
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/State/Templates/DeckBuilderState.cs]: I still don't understand this method.

Why is this returning a type of `Deck`?

It looks like the intent is to get all possible cards that could be in the deck. So, is this supposed to be named `GetPossibleDeckCardsInCollection`, which should return a `List<Card>`?

Not all collections of cards are `Decks`.

**Pull request merged**